File: SDKs/amazonaws.com/servicediscovery/2017-03-14/python/src/sdk/models/shared/privatednsnamespacepropertieschange.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import privatednspropertiesmutablechange as shared_privatednspropertiesmutablechange


@dataclass_json
@dataclasses.dataclass
class PrivateDNSNamespacePropertiesChange:
    r"""PrivateDNSNamespacePropertiesChange
    Updated properties for the private DNS namespace.
    """
    
    dns_properties: shared_privatednspropertiesmutablechange.PrivateDNSPropertiesMutableChange = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('DnsProperties') }})
    

File: SDKs/amazonaws.com/servicediscovery/2017-03-14/java/lib/src/main/java/openapisdk/models/shared/PrivateDnsNamespacePropertiesChange.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * PrivateDnsNamespacePropertiesChange
 * Updated properties for the private DNS namespace.
**/
public class PrivateDnsNamespacePropertiesChange {
    @JsonProperty("DnsProperties")
    public PrivateDnsPropertiesMutableChange dnsProperties;
    public PrivateDnsNamespacePropertiesChange withDnsProperties(PrivateDnsPropertiesMutableChange dnsProperties) {
        this.dnsProperties = dnsProperties;
        return this;
    }
}